app-shells/hstr/patches/hstr-1.24.patchset

```diff
@@ -0,0 +1,86 @@
+From 419c19184f486d849f845e8485fd552d4c83b51b Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Zolt=C3=A1n=20Mizsei?= <zmizsei@extrowerk.com>
+Date: Sun, 29 Apr 2018 12:58:42 +0200
+Subject: Haiku supporting patches
+
+
+diff --git a/src/Makefile.am b/src/Makefile.am
+index c3e0823..20f8e44 100644
+--- a/src/Makefile.am
++++ b/src/Makefile.am
+@@ -34,6 +34,6 @@ hh_SOURCES = 						\
+ 
+ # create hh > hstr hard link on installation
+ install-exec-hook:
+-	ln $(DESTDIR)$(bindir)/hh$(EXEEXT) $(DESTDIR)$(bindir)/hstr$(EXEEXT)
++	ln -s $(DESTDIR)$(bindir)/hh$(EXEEXT) $(DESTDIR)$(bindir)/hstr$(EXEEXT)
+ 
+ hh_LDADD = $(NCURSES_LIBS)
+diff --git a/src/hstr.c b/src/hstr.c
+index 80b18d2..5141663 100644
+--- a/src/hstr.c
++++ b/src/hstr.c
+@@ -22,6 +22,8 @@
+ #include <locale.h>
+ #ifdef __APPLE__
+ #include <curses.h>
++#elif defined(__HAIKU__)
++#include <ncurses.h>
+ #elif defined(__FreeBSD__)
+ #include <ncurses.h>
+ #else
+diff --git a/src/hstr_history.c b/src/hstr_history.c
+index 1e01f35..fff0549 100644
+--- a/src/hstr_history.c
++++ b/src/hstr_history.c
+@@ -55,7 +55,11 @@ char *get_history_file_name()
+     if(!historyFile || strlen(historyFile)==0) {
+         char *home = getenv(ENV_VAR_HOME);
+         historyFile = malloc(strlen(home) + 1 + strlen(FILE_DEFAULT_HISTORY) + 1);
+-        strcat(strcat(strcpy(historyFile, home), "/"), FILE_DEFAULT_HISTORY);
++        #ifndef __HAIKU__
++            strcat(strcat(strcpy(historyFile, home), "/"), FILE_DEFAULT_HISTORY);
++        #else
++            strcat(strcat(strcpy(historyFile, home), "/config/settings/"), FILE_DEFAULT_HISTORY);
```

Comment:
Shall we instead use [`find_directory()`][0]?

[0]: https://www.haiku-os.org/docs/api/FindDirectory_8h.html#af95c55d472448ab7af9f32c33aeea71e